test/virtual_dataset.jl

```diff
@@ -5,36 +5,36 @@ using Test, HDF5
 
     filename = joinpath(dirname, "main.hdf5")
 
-    f = h5open(filename, "w")
```

Comment:
The issue was here. We forgot to close this handle explicitly.

Reply:
Probably best to pull this into a different PR

Reply:
I can pull this into a different pull request, but this needs to be here as well since CI will error otherwise.